repos/fdbt-site/src/pages/api/updateFareTriangle.ts

```diff
@@ -0,0 +1,158 @@
+import { NextApiResponse } from 'next';
+import { putDataInS3 } from '../../data/s3';
+import { NextApiRequestWithSession, ErrorInfo, UserFareStages } from '../../interfaces';
+import { redirectToError, redirectTo } from '../../utils/apiUtils';
+import {
+    INPUT_METHOD_ATTRIBUTE,
+    CSV_UPLOAD_ATTRIBUTE,
+    MATCHING_JSON_ATTRIBUTE,
+    PRODUCT_AND_SERVICE_ID_ATTRIBUTE,
+} from '../../constants/attributes';
+import { getFormData, processFileUpload } from '../../utils/apiUtils/fileUpload';
+import { getSessionAttribute, updateSessionAttribute } from '../../utils/sessions';
+import { WithIds, ReturnTicket, SingleTicket } from '../../../shared/matchingJsonTypes';
+import { faresTriangleDataMapper, setCsvUploadAttributeAndRedirect } from './csvUpload';
+
+const errorId = 'csv-upload';
+
+export const config = {
+    api: {
+        bodyParser: false,
+    },
+};
+
+const getNamesOfFareZones = (ticket: WithIds<SingleTicket> | WithIds<ReturnTicket>) => {
+    let fareZoneNames;
+
+    if ('fareZones' in ticket) {
+        fareZoneNames = ticket.fareZones.map((fz) => fz.name);
+    } else if ('inboundFareZones' in ticket && 'outboundFareZones' in ticket) {
+        const combinedFareZones = ticket.outboundFareZones.concat(ticket.inboundFareZones);
+
+        const combinedNamesOnly = combinedFareZones.map((x) => x.name);
+
+        // get rid of duplicates
+        fareZoneNames = [...new Set(combinedNamesOnly)];
+    }
+
+    return fareZoneNames;
+};
+
+const checkFareStageNames = (fareTriangleData: UserFareStages, fareZoneNames: string[]): boolean => {
+    const thereIsANameMismatch = fareTriangleData.fareStages.some((fs) => {
+        return !fareZoneNames.includes(fs.stageName);
+    });
+
+    return thereIsANameMismatch;
+};
+
+const getStageCountMismatchError = () => {
```

Comment:
Does this and the one below need to be a function? Could it instead just be a variable? 
```suggestion
const stageCountMismatchError: ErrorInfo[] = [
        {
            id: errorId,
            errorMessage:
                'The number of fare stages of your updated fares triangle do not match the one you have previously uploaded. Update your triangle, ensuring the number of fare stages match before trying to upload again',
        },
    ];
```